unipop-elastic/src/org/unipop/elastic/controller/schema/SchemaVertex.java

```diff
@@ -0,0 +1,103 @@
+package org.unipop.elastic.controller.schema;
+
+import com.google.common.collect.FluentIterable;
+import org.apache.tinkerpop.gremlin.structure.Property;
+import org.apache.tinkerpop.gremlin.structure.VertexProperty;
+import org.unipop.elastic.controller.schema.helpers.schemaProviders.GraphElementSchemaProvider;
+import org.unipop.elastic.controller.schema.helpers.schemaProviders.GraphVertexSchema;
+import org.unipop.elastic.helpers.ElasticMutations;
+import org.unipop.elastic.helpers.LazyGetter;
+import org.unipop.structure.BaseVertex;
+import org.unipop.structure.BaseVertexProperty;
+import org.unipop.structure.UniGraph;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * Created by Roman on 9/21/2015.
+ */
+public class SchemaVertex<TController extends SchemaVertexController> extends BaseVertex<TController> {
+    //region Constructor
+    public SchemaVertex(
+            Object id,
+            String label,
+            UniGraph graph,
+            Map<String, Object> keyValues,
+            TController vertexController,
+            LazyGetter lazyGetter,
+            GraphElementSchemaProvider schemaProvider,
+            ElasticMutations elasticMutations) {
+        super(id, label, keyValues, vertexController, graph);
+        this.schema = schemaProvider.getVertexSchema(this.label());
+
+        this.elasticMutations = elasticMutations;
+        if (lazyGetter != null) {
+            this.lazyGetter = lazyGetter;
+            lazyGetter.register(this, label, this.schema.get().getIndices().iterator().next());
+        }
+    }
+    //endregion
+
+    //region BaseVertex Implementation
+    @Override
+    protected void innerAddProperty(BaseVertexProperty vertexProperty) {
+        try {
+            String writeIndex = FluentIterable.from(schema.get().getIndices()).first().get();
```

Comment:
The project isn't compiling because we don't have a dependeny on Guava for FluentIterable.
Could you change the usage to Java8 Streams? Because the rest of the code uses that.
StreamSupport.stream(schema.get().getIndices().spliterator(), false).findFirst().get();
(If you save stuff as Collections it'll just be "schema.get().getIndices().stream().findFirst().get()")